### ISSUE TITLE
Bug: Header Nav Colors Updating Incorrectly in V3-Beta.110 Website Template

### ISSUE THREAD

**Issue by MotorcycleEnjoyer** [payloadcms/payload#8529]
### Environment Info

Binaries:
  Node: 20.12.2
  npm: N/A
  Yarn: N/A
  pnpm: 9.1.0
Relevant Packages:
  payload: 3.0.0-beta.110
  next: 15.0.0-canary.160
  @payloadcms/db-mongodb: 3.0.0-beta.110
  @payloadcms/email-nodemailer: 3.0.0-beta.110
  @payloadcms/graphql: 3.0.0-beta.110
  @payloadcms/live-preview: 3.0.0-beta.110
  @payloadcms/live-preview-react: 3.0.0-beta.110
  @payloadcms/next/utilities: 3.0.0-beta.110
  @payloadcms/plugin-cloud: 3.0.0-beta.110
  @payloadcms/plugin-form-builder: 3.0.0-beta.110
  @payloadcms/plugin-nested-docs: 3.0.0-beta.110
  @payloadcms/plugin-redirects: 3.0.0-beta.110
  @payloadcms/plugin-search: 3.0.0-beta.110
  @payloadcms/plugin-seo: 3.0.0-beta.110
  @payloadcms/richtext-lexical: 3.0.0-beta.110
  @payloadcms/translations: 3.0.0-beta.110
  @payloadcms/ui/shared: 3.0.0-beta.110
  react: 19.0.0-rc-5dcb0097-20240918
  react-dom: 19.0.0-rc-5dcb0097-20240918
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
  Available memory (MB): 32631
  Available CPU cores: 24

### Describe the Bug

After seeding the database on `V3 Beta 110 website template`, the following problem exists:

On the Homepage you cannot see the search icon because it stays black, even on page refresh.

![Image](https://github.com/user-attachments/assets/e1a7426b-04f4-4210-ba0a-97b93d9a5942)

If you click on "posts" or "all posts" to navigate to `/posts` page, the Nav colors do not update.

![Image](https://github.com/user-attachments/assets/3b3b1aa8-f87d-4430-a6ea-1ca8a04689f6)

While on the `/posts` page if you refresh `F5`, the colors update correctly.
While on the `/posts` page if you navigate back to home by clicking on back arrow or payload icon, the colors update correctly.

`/contact` has same behavior as `/posts`

TLDR: 
- Nav Search icon does not change colors (stays black).
- Nav Logo and links do not change color when navigating away from homepage

### Reproduction Steps

1.) Create V3 Website Template with pnpm and mongodb `npx create-payload-app@beta --use-pnpm`
2.) run `pnpm dev`
3.) Visit localhost:3000/admin
4.) Create a user
5.) Seed the database
6.) Visit `Home` page after seeding database
7.) Press `F5` and notice search icon does not change color
8.) Click `posts` nav link after visiting `Home` and notice colors not updating
9.) Press `F5` and notice the colors updating correctly on nav except search icon
10.) Click on the Payload logo to navigate to `Home`
11.) Notice the colors updating correctly except search icon

Browers tested:
Firefox: 130.0.1 (64-bit)
Chrome: 129.0.6668.71 (Official Build) (64-bit)

**Comment by paulpopus** [comment]
Thanks for bringing this my attention! Fixed in https://github.com/payloadcms/payload/pull/8542
